File: Data Structures & Algorithms/house-robber-ii/submission-3.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int dp1[101];
        int dp2[101];
        int n = nums.size();

        if(n==1)
            return nums[0];
        
        dp1[0] = nums[0];
        dp2[0] = 0;
        if(n>1){
            dp1[1] = max(nums[0],nums[1]);
            dp2[1] = nums[1];
        }

        for(int i=2;i<n;i++){
            dp1[i] = max(dp1[i-1],dp1[i-2]+nums[i]);
            if(n==2)
                dp2[2] = max(nums[2],nums[1]);
            else
                dp2[i] = max(dp2[i-1],dp2[i-2]+nums[i]);
        }

        return max(dp1[n-2],dp2[n-1]);
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int max = 0;

        HashSet<Integer> set = new HashSet<>();

        for(int i : nums){
            set.add(i);
        }

        for(int i : set){
            if(!set.contains(i-1)){
                int curr = i;
                int count=1;

                while(set.contains(curr+1)){
                    curr++;
                    count++;
                }
                max = Math.max(count,max);
            }
        }
        return max;
    }
}
